data/src/main/java/com/my/kizzy/data/remote/ApiService.kt

```diff
@@ -52,6 +57,75 @@ class ApiService @Inject constructor(
                 }
             ))
         }
+    }.getOrNull()?.toImageAsset()
+
+    open suspend fun HttpResponse.toImageAsset(): String? {
+        return try {
+            if (this.status == HttpStatusCode.OK)
+                this.body<ApiResponse>().id
+            else
+                null
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    class ImgurApiService @Inject constructor(
+        private val client: HttpClient,
+        @Discord private val discordBaseUrl: String,
+        @Imgur private val imgurBaseUrl: String,
+    ) : ApiService(client, "", discordBaseUrl, "") {
+        override suspend fun getImage(url: String, token: String) = runCatching {
+            client.post {
+                url("$discordBaseUrl/applications/$APPLICATION_ID/external-assets")
+                headers {
+                    append(HttpHeaders.Authorization, token)
+                    append(HttpHeaders.ContentType, "application/json")
+                }
+                setBody(mapOf("urls" to arrayOf(url)))
+            }
+        }.getOrNull()?.toExternalImage()
+
+        override suspend fun uploadImage(file: File, token: String) = runCatching {
+            client.post {
+                url("$imgurBaseUrl/image")
+                headers {
+                    // Imgur web client API key, unchanged for at least >5 years as of 2024
+                    append(HttpHeaders.Authorization, "Client-ID 546c25a59c58ad7")
+                }
+                contentType(ContentType.MultiPart.FormData)
+                setBody(
+                    MultiPartFormDataContent(
+                        formData {
+                            append("image", file.readBytes())
+                            append("type", "raw")
+                        }
+                    )
+                )
+            }
+        }.getOrNull()?.toImageAsset()?.let { this.getImage(it, token) }
+
+        override suspend fun HttpResponse.toImageAsset(): String? {
+            return try {
+                if (this.status == HttpStatusCode.OK)
+                    this.body<ImgurResponse>().data.link
+                else
+                    null
+            } catch (e: Exception) {
+                null
+            }
+        }
+
+        suspend fun HttpResponse.toExternalImage(): String? {
+            return try {
+                if (this.status == HttpStatusCode.OK)
+                    "mp:" + this.body<Array<ExternalAsset>>().first().externalAssetPath
+                else
+                    null
+            } catch (e: Exception) {
+                null
+            }
+        }
```

Comment:
No need to extend ApiService. We can have ImgurApiService as a standalone class. Also put it in another file